asset/interface.go

```diff
@@ -24,3 +28,28 @@ type GenesisTxBuilder interface {
 	// output is used to create a group witness for the grouped asset.
 	BuildGenesisTx(newAsset *Asset) (*wire.MsgTx, *wire.TxOut, error)
 }
+
+// TapscriptTreeStore is used to persist a Tapscript tree associated with an
+// internal key. The resulting tweaked key can be an asset script key, group
+// key, or on-chain Taproot output key. The internal key does not need to be
+// stored prior to storing the associated Tapscript tree. The associated
+// Tapscript tree cannot be updated once stored.
+type TapscriptTreeStore interface {
+	// StoreTapscriptTree persists a Tapscript tree associated with an
+	// internal key. Storing an empty Tapscript tree is a no-op, and storing
+	// a Tapscript tree for a key that already has a stored Tapscript tree
+	// returns a concrete error. The internal key for the control block of
+	// the Tapscript object must be set to the internal key.
+	StoreTapscriptTree(ctx context.Context,
+		tapTree TapscriptTreePreimage) (*chainhash.Hash, error)
```

Comment:
Return value being the root hash here?

Reply:
Yes exactly, which would then be added to the batch struct and stored in the DB as part of the batch itself.